compiler/qsc_fir/src/fir.rs

```diff
@@ -1423,6 +1423,89 @@ impl Display for PatKind {
     }
 }
 
+/// A [`VecIdent`] represents a sequence of idents. It provides a helpful abstraction
+/// that is more powerful than a simple `Vec<Ident>`, and is primarily used to represent
+/// dot-separated paths.
```

Comment:
This looks familiar :-/

Reply:
Do we really need the vector representation in the FIR though? One advantage of lowering is that you get to abstract away details that don't matter at this level. Keeping the parts separately in a vector is useful when you're doing name resolution, but once we're in the FIR we're done with all that. We can keep the whole thing as a single string.

IMO this means you can revert all the changes to the `qsc_fir` folder and do away with `Idents` at the lowering step. Like this: https://github.com/microsoft/qsharp/commit/bed6b712133b0b687fd4cf34ac1ea5c20cda9858